Src/Plugins/ValidatorBase.cs

```diff
@@ -199,6 +214,50 @@ public static string ParseExpression(IRegex regexEngine, string matchedPattern,
             return match?.Success ?? false ? ParseValue(match.Value) : null;
         }
 
+        public static void StandardizeLocalhostName(Dictionary<string, string> groups, string hostKey = "host")
+        {
+            if (groups.TryGetNonEmptyValue(hostKey, out string host))
+            {
+                if (LocalhostList.Contains(host))
+                {
+                    groups[hostKey] = "localhost";
+                }
+            }
+        }
+
+        public virtual string HostExclusion(ref Dictionary<string, string> groups, IEnumerable<string> hostList = null, string hostKey = "host")
```

Comment:
It's making me uncomfortable that this very low-level class has handling built into it for a topic (host management) that won't be relevant to all derived types. Let me think on this overnight. :)

Reply:
I'm not married to keeping this code here.  I just saw it in two validators and wanted to avoid the repetition.  Do you think a utility class might be a more appropriate location?

---
In reply to: [574950963](https://github.com/microsoft/sarif-pattern-matcher/pull/189#discussion_r574950963) [](ancestors = 574950963)

Reply:
Here is my suggestion:

1. Author a new base class that extends validator base, call it 'LocalHostFilteringValidator' or 'DomainFilteringValidator' or some such. Mark this type as abstract.
2. Add your helpers, etc. to the new base type. Add the actual domain filtering to the IsValidStaticHelper override.
3. Update all relevant validators to extend the new base type. Update their IsValidStaticHelper overrides to first call the base type. If the base type indicates 'NoMatch', indicating that a match was filtered due to the domain encountered in the host data, stop processing. Otherwise the child type continue w/its own static analysis.


---
In reply to: [575386331](https://github.com/microsoft/sarif-pattern-matcher/pull/189#discussion_r575386331) [](ancestors = 575386331,574950963)

Reply:
Moved most of the code, I kept "MatchCleanup" here because all validators have the potential to need to "cleanup" their matches.  it is not a domain specific thing.  refactored MySql, Sql, and PostgreSql.  Let me know what you think!

---
In reply to: [575421009](https://github.com/microsoft/sarif-pattern-matcher/pull/189#discussion_r575421009) [](ancestors = 575421009,575386331,574950963)

Reply:
Fixed as discussed.

---
In reply to: [575475731](https://github.com/microsoft/sarif-pattern-matcher/pull/189#discussion_r575475731) [](ancestors = 575475731,575421009,575386331,574950963)